### PR TITLE
Add support for the nnpchecker option

### PR DESCRIPTION
**Pull request by kit-ty-kate**

### CODE REVIEW

File: ocaml_version.ml [hunk 1]
```diff
@@ -311,6 +312,7 @@ module Configure_options = struct
     | `Force_safe_string -> "force safe string mode"
     | `Frame_pointer -> "frame pointer"
     | `No_naked_pointers -> "forbid unboxed pointers"
+    | `No_naked_pointers_checker -> "forbid unboxed pointers and print a warning"
```
Comment (dra27) [inline ocaml_version.ml:315]:
```suggestion
    | `No_naked_pointers_checker -> "enable the naked pointers checker"
```
(naked pointers are still allowed - it actually conflicts `` `No_naked_pointers``)

File: ocaml_version.ml [hunk 2]
```diff
@@ -419,6 +424,12 @@ let compiler_variants arch ({major; minor; _} as t) =
       (* No variants for OCaml < 4.00 *)
       if version < (4, 00) then []
       else
+        (* +nnpchecker for OCaml 4.12+ *)
+        let variants =
+          if version >= (4, 12) then
```
Comment (dra27) [inline ocaml_version.ml:429]:
```suggestion
        (* +nnpchecker for OCaml 4.12+ on x86_64 *)
        let variants =
          if arch = `X86_64 && version >= (4, 12) then
```

**Review by dra27 — COMMENTED** [review]
Two minor tweaks

**Comment by dra27** [comment]
Thanks!

**Comment by avsm** [comment]
For the future, please try not to merge to the main branch until it's run through a successful build somewhere, or it makes updating vendored copies of the library tricky. This was broken on two fronts:

- trunk-variants needs to be x86_64 only so the 4.13 build failed (fixed in a4560fbc68012aa9c98af6b51f7f7aad0ee4c775)
- there was no ocaml-options-only-nnpchecker package so the 4.12 build failed (fixed in https://github.com/ocaml/opam-repository/pull/18258)

Once these are in, I'll push an update to base-images again.

**Pull request merged**